File: Java/Week 3/KthDigitWithPower.java
// Finds the power of A power B and then finds the kth digit of the number
class KthDigitWithPower {
    public static void main(String[] args) {
       int A = 3, B=3, k=1;
      // Built-in pow function to find A power B
       long power=(long)Math.pow(A,B);
        long i=1;
        
        while(power>0){
            long ans=power%10;
          // 
            if(k==i){
                return ans;
            }
            i++;
            power=power/10;
        }
        return 0;
       
    }
}



// In online compilers use the below code

class Main {
    public static void main(String[] args) {
       int A = 3, B=3, k=1;
       long power=(long)Math.pow(A,B);
        long i=1;
        
        while(power>0){
            long ans=power%10;
            if(k==i){
                System.out.println(ans);
                break;
            }
            i++;
            power=power/10;
        }
        
       
    }
}

File: Cpp/Week 3/kthDigitWithPower.cpp
// In GeekforGeeks
https://www.geeksforgeeks.org/problems/print-the-kth-digit3520/0
class Solution{
public:
    int kthDigit(int A,int B,int k){
        // code here
        long long power=pow(A,B);
        int i=1;
        
        while(power>0){
            int ans=power%10;
            if(k==i){
                return ans;
            }
            i++;
            power=power/10;
        }
        return 0;
    }
};

In Online compilers 

#include <iostream>
#include<math.h>
using namespace std;
int main() {
    int A=3, B=3, k=1;
    long long power=pow(A,B);
        int i=1;
        
        while(power>0){
            int ans=power%10;
            if(k==i){
                cout<< ans;
            }
            i++;
            power=power/10;
        }
    return 0;
}
